x-pack/plugins/apm/server/routes/apm_routes/register_apm_server_routes.ts

```diff
@@ -80,7 +105,7 @@ export function registerRoutes({
         options,
         validate: routeValidationObject,
       },
-      async (context, request, response) => {
+      wrapRouteHandlerInProfiler(async (context, request, response) => {
```

Comment:
Is there any issue raised for this? (being able to profile apm requests)

Reply:
No, it's for development purposes.

Reply:
LGTM if this is something that we need

Reply:
To clarify, I have added this on my own initiative - I think in many cases we are blind to CPU usage in our endpoints. For most of our requests, it's not an issue, but for heavy processing endpoints like the one that downloads an entire trace, or the one that aggregates critical path, I think it's helpful.